pkg/sources/git/git.go

```diff
@@ -371,6 +383,7 @@ func (s *Git) ScanCommits(ctx context.Context, repo *git.Repository, path string
 			break
 		}
 		depth++
+		s.metrics.commitsScanned[commit.Hash] = struct{}{}
```

Comment:
This seems like it would use a lot of memory for large repos. Just `trufflehog` would be 88KB (barring the overhead of maps).

Reply:
Good point. I used the method from thog without really considering if it was the best. Switched to just counting since thats all that is used.